### ISSUE TITLE
How to get this to work within a slider?

### ISSUE THREAD

**Issue by jamminjames** [timmywil/jquery.panzoom#193]
I can't seem to make this work within a slider I'm using. I have the image class set to "panzoom" and the scripts loaded, and the following script before the /body tag:
```define([ "jquery", "plugins/jquery.panzoom" ], function( $ ) {
 $(document).ready(function() {
$(".panzoom-elements").panzoom();
// Pass options
$("a.panzoom-elements").panzoom({
  minScale: 0,
  $zoomRange: $("input[type='range']")
});
});
});```
What am I missing? Or will it not work within a jquery slider? (I'm using the old wmuSlider.)

**Comment by newcurzon** [comment]
i have experience same like you, im using swiper and panzoom, after doubleclick or pinch in will call panzoom, after pinch out or second doubleclick call panzoom('destroy') then my swiper can work properly.

**Comment by spnfury** [comment]
One more...

**Comment by willcalderbank** [comment]
+1

**Comment by willcalderbank** [comment]
There is a work around, you can disable the panzoom and use hammer.js to zoom/pan it.

```
onZoomChanged: function(zoom) {

    if (zoom === 1) {
        $scope.zoom.api.disable();
    } else {
        $scope.zoom.api.enable();
    }
}


$scope.zoom.api.disable();

_zoomMc = new Hammer(
    $('.zoomable-image-wrapper-'+$scope.$index).get(0)
);
_zoomMc.add( new Hammer.Pinch() );

_zoomMc.on('pinchstart', function(ev) {
    startDistance = getTouchDistance(ev.pointers);
    startX = ev.center.x - $('.zoomable-image-wrapper-'+$scope.$index).width()/2;
    startY = ev.center.y - $('.zoomable-image-wrapper-'+$scope.$index).height()/2;
});

_zoomMc.on('pinch', function(ev) {

    $scope.zoom.api.enable();
    var matrix = $scope.zoom.api.getMatrix();


    $scope.zoom.api.zoomTo(
        ((getTouchDistance(ev.pointers) - startDistance) * 0.005) + 1,
        { animate: false, matrix: $scope.zoom.api.getMatrix() }
    );

    $scope.zoom.api.pan(
        ev.center.x - $('.zoomable-image-wrapper-'+$scope.$index).width()/2 - startX,
        ev.center.y - $('.zoomable-image-wrapper-'+$scope.$index).height()/2 - startY,
        { matrix: $scope.zoom.api.getMatrix(), animate: 'skip' }
    );

});
```

Apologies for the none-cut-and-pastable code, we've wrapped panzoom in angular.

**Comment by brucewyne** [comment]
FWIW. I just came across this today. Setting disablePan: true lets swiper.js work just fine. But obviously you want panning. So I settled that I only want the swiper when the image is at the normal size.  

`$(".zoom-image").panzoom({
	disablePan: true
});
$(".zoom-image").on('panzoomzoom',function(e,panzoom,scale){
	if(scale==1){
		panzoom.options.disablePan = true;
	} else {
		panzoom.options.disablePan = false;
	}
});
// reset zoomed product
$('.zoom-reset').click(function () {
	$(".zoom-image").panzoom('zoom',1);
});`
